### DISCUSSION TITLE
Problems with quickstart example in ubuntu 22.04. It works, but I do not get the figures

### DISCUSSION THREAD

**Discussion by ricguerr** [georgebv/pyextremes#52] [Q&A] [answered]
Dear all:

I am getting myself familiar with pyExtremes and I am trying to run the quickstart example.
I have installed pyExtremes with "pip install pyextremes[full]" in a clean virtual environment in my ubuntu 22.04 machine. pip installs several dependencies, as shown:

contourpy==1.0.7
cycler==0.11.0
emcee==3.1.4
fonttools==4.39.3
kiwisolver==1.4.4
matplotlib==3.7.1
numpy==1.24.3
packaging==23.1
pandas==2.0.1
Pillow==9.5.0
pyextremes==2.3.0
pyparsing==3.0.9
python-dateutil==2.8.2
pytz==2023.3
scipy==1.10.1
six==1.16.0
tqdm==4.65.0
tzdata==2023.3

I can run the example, and it works properly because I can print the summary. But I am not getting the figures.

¿Can some one explain me what is the problem?. I thought that I was missing graphics libraries, so I also installed PyQt5 with pip, but after that I am still having the same problem

Thank you for your advice

**Comment by ricguerr** [comment]
I solved partially the problem adding plt.show() at the end of 
    def plot_diagnostic( in extremes.py
and
   def plot_extremes( in eva.py

although with a blocking behaviour. plt.show(blocking=False) do not works as creates an empty figure-

But I imagine this is not a solution.

¿Which is the suggested way to run pyExtremes? Inside a Jupyter Notebooks?

I have missinterpretated a previos question thinking that in that question the problem was about the threshold limit. It was similar to mine. But in my case, the trick with main did not worked

**Reply by georgebv** [reply, marked as answer]
You should look at return types for those functions - those are matplotlib figures and axes. If you are in non-interactive environment, then you should save those figures to a file and then explicitly close the figure instance. I recommend reading this https://matplotlib.org/stable/users/explain/interactive.html
